sdk-workflows/src/main/java/io/dapr/workflows/Workflow.java

```diff
@@ -30,10 +34,49 @@ public Workflow(){
   /**
    * Executes the workflow logic.
    *
-   * @param ctx provides access to methods for scheduling durable tasks and getting information about the current
+   * @param ctx provides access to methods for scheduling durable tasks and
+   *            getting information about the current
    *            workflow instance.
    */
   public void run(WorkflowContext ctx) {
-    this.create().run(ctx);
+    WorkflowStub stub = this.create();
+
+    if (!this.isSagaEnabled()) {
+      // saga disabled
+      stub.run(ctx);
+    } else {
+      // saga enabled
+      try {
+        stub.run(ctx);
+      } catch (OrchestratorBlockedException e) {
```

Comment:
Besides `OrchestratorBlockedException`, there is another exception [`ContinueAsNewInterruption `](https://github.com/microsoft/durabletask-java/blob/main/client/src/main/java/com/microsoft/durabletask/interruption/ContinueAsNewInterruption.java) that used as an interruption when customer call `continueAsNew`, it interrupts the workflow and yields control back to the workflow Grpc server. Should we also re-throw it here? 

I think it will be a good test case to add for combining `continueAsNew` and Saga option.

Reply:
Thanks for your reminder, this exception must be handled the same way. I will update the proposal and code.